Source/WebCore/css/parser/CSSPropertyParserConsumer+FlexDefinitions.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2024 Samuel Weinig <sam@webkit.org>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "CSSPropertyParserConsumer+MetaConsumerDefinitions.h"
+
+namespace WebCore {
+namespace CSSPropertyParserHelpers {
+
+struct FlexValidator {
+    static bool isValid(CSSUnitType unitType, CSSPropertyParserOptions)
+    {
+        switch (unitType) {
+        case CSSUnitType::CSS_FR:
+            return true;
+
+        default:
+            return false;
+        }
+    }
+
+    template<auto R> static bool isValid(CSS::FlexRaw<R> raw, CSSPropertyParserOptions)
+    {
+        return isValidCanonicalValue(raw);
+    }
+};
+
+template<auto R> struct ConsumerDefinition<CSS::Flex<R>> {
+    using FunctionToken = FunctionConsumerForCalcValues<CSS::Flex<R>>;
+    using DimensionToken = DimensionConsumer<CSS::Flex<R>, FlexValidator>;
+
+    // NOTE: Flex does not allow unitless consumers.
```

Comment:
Does this comment mean it should not, but currently does?

Reply:
It is supposed to mean that it both does not and should never (only angle and length should allow unitless). I’ll make it a bit more clear.